spec/controllers/conference_registration_controller_spec.rb

```diff
@@ -0,0 +1,284 @@
+require 'spec_helper'
+
+describe ConferenceRegistrationsController, type: :controller do
+  let!(:first_user) { create(:user) }
+  let(:conference) { create(:conference) }
+  let(:conference_with_open_registration) { create(:conference) }
+  let!(:open_registration_period) { create(:registration_period, conference: conference_with_open_registration, start_date: Date.current - 6.days) }
+  let(:user) { create(:user) }
+
+  context 'user is not signed in' do
+    describe 'GET #new' do
+      context 'ichain is enabled' do
+        before do
+          CONFIG['authentication']['ichain']['enabled'] = true
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        after { CONFIG['authentication']['ichain']['enabled'] = false }
+
+        it 'redirects to root path' do
+          expect(response).to redirect_to root_path
+        end
+      end
+    end
+
+    describe 'POST #create' do
+      before do
+        post :create, user: attributes_for(:user),
+                      registration: attributes_for(:registration),
+                      conference_id: conference_with_open_registration.short_title
+      end
+
+      it 'assigns user variable' do
+        expect(assigns(:user)).not_to be_nil
+      end
+
+      it 'signs in registration user' do
+        expect(controller.current_user).not_to be_nil
+      end
+
+      it 'shows success message in flash notice' do
+        expect(flash[:notice]).to match('You are now registered and will be receiving E-Mail notifications.')
+      end
+
+      it 'redirects to registration show path' do
+        expect(response).to redirect_to conference_conference_registrations_path(conference_with_open_registration.short_title)
+      end
+
+      it 'creates a new registration' do
+        expect(Registration.count).to eq 1
+      end
+    end
+  end
+
+  context 'user is signed in' do
+    before { sign_in(user) }
+
+    describe 'GET #new' do
+      context 'user is registered to conference' do
```

Comment:
So psuedo code would be `new` would be something like this:
```
if user is registered
   # do something
elsif ichain is enabled and user is not signed in
  # do something else
end

if user is signed in and registerd
   # do something
end

if registration limit has reached
   # do something
end

# do something if none of the above if blocks already redirect request 
```
So the way things are set up, there really isn't a context for `user is not registered`. [context](https://github.com/openSUSE/osem/pull/795/files#diff-f594354d23f55f0c31fc2d9fbd49250eR86) of `successful request` probably the closet thing.